### PR TITLE
fix up meta.yml and generate README, opam, and GitHub Actions CI

### PR DESCRIPTION
**Pull request by palmskog**
@amahboubi here is my idiomatic version of `meta.yml`, with all the generated files. Feel free to comment directly on the PR with things that were not clear to you.

### CODE REVIEW

**Comment by chdoc** [comment]
Shouldn't this already trigger a CI run?

**Comment by palmskog** [comment]
There is one here: https://github.com/palmskog/apery/runs/1473477103

**Comment by chdoc** [comment]
Yes, that's your "branch CI", but shouldn't there also be CI runs on this side, or is there something that needs activating? in the settings of the organization or the repository?

**Comment by palmskog** [comment]
The repo currently has no `.github` directory, so I guess Actions CI doesn't activate when the PR is from some other repo.

**Comment by chdoc** [comment]
That might be it, because I'm pretty sure it did activate immediately when I made "local" branches in my coq-community repositories.

**Comment by chdoc** [comment]
@amahboubi @palmskog Is apery known to break with `coq-8.12 ` and/or `mathcomp-1.11` ?

**Comment by palmskog** [comment]
@chdoc yes, it only works with MathComp 1.10

**Comment by amahboubi** [comment]
Sorry, I was not available for this yesterday. I am back on tracks.

**Comment by amahboubi** [comment]
@palmskog thanks a lot for this PR. As expected, I am not competent to review it so I will ask my (newbie) questions as comments in your code, as you suggest.

**Comment by amahboubi** [comment]
And my first question is: have files `coq-ci.yml`, `README.ml` and `coq-mathcomp-apery.opam` been generated from `meta.yml` or did you edited them manually?

**Comment by chdoc** [comment]
@amahboubi These should all be generated from `meta.yml`.

File: meta.yml [hunk 1]
```diff
@@ -33,36 +33,52 @@ authors:
...
+  text: 8.11
+  opam: '{>= "8.11" & < "8.12~"}'
 
 tested_coq_opam_versions:
-- version: '8.11'
+- version: '1.10.0-coq-8.11'
+  repo: 'mathcomp/mathcomp'
 
 dependencies:
+- opam:
```
Comment (amahboubi) [inline meta.yml:54]: Should I provide explicitly the description of all dependencies? Is it (only) for a better looking README?

File: README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Apery
 
-[![Travis][travis-shield]][travis-link]
+[![CI][action-shield]][action-link]
```
Comment (amahboubi) [inline README.md:3]: This diff if very mysterious to me. Which change in the config files did trigger it? What are the consequences?

File: meta.yml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 fullname: Apery
-shortname: mathcomp-apery
+shortname: apery
 organization: math-comp
+opam_name: coq-mathcomp-apery
```
Comment (amahboubi) [inline meta.yml:4]: We had some discussion at an earlier Mathcomp Dev [WG](https://github.com/math-comp/math-comp/wiki/Minutes-June-16-2020) about opam naming policy which was not conclusive. If I understand correctly, the hard constraint is a `coq-` prefix. Then the question is `coq-mathcomp-apery` or `coq-apery`...

File: meta.yml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 community: false
-travis: true
+action: true
```
Comment (amahboubi) [inline meta.yml:6]: Ha, I guess the answer to my previous comment about the mysterious diff is here. So on which basis should I choose between `action` and `travis`?

Comment (chdoc) [reply meta.yml:6]: Travis started severely limiting (on a per-organization basis) the amount of computation time given to non-paying open-source projects. Unless you have a strong reason to use Travis for one of its features, don't.

File: coq-mathcomp-apery.opam [hunk 5]
```diff
@@ -1,13 +1,13 @@
 opam-version: "2.0"
 maintainer: "assia.mahboubi@inria.fr"
-version: "released"
+version: "dev"
```
[the remark this answers is not in the document]
Comment (chdoc) [reply coq-mathcomp-apery.opam:3]: IIUC, `released` is not a version. Opam versions should either be `dev` for the the latest in the repository or numeric `1.0` for a released snapshot. So the one inside the repository used for opam-based CI should always be `dev`.

File: README.md (hunk 2 above)
Comment (CohenCyril) [reply README.md:3]: This is just a "badge" update

Comment (amahboubi) [reply README.md:3]: My question was actually about travis vs ci,  not about the badges themselves. But I got the answer below.

**Pull request merged**